packages/cli/src/commands/alter.ts

```diff
@@ -166,13 +167,48 @@ export async function alter(
 
         // some steps may require access to misc artifacts
         await runtime.restoreMisc(deployInfo.miscUrl);
-        deployInfo.state[stepName].artifacts = await stepAction.importExisting(
+
+        const importExisting = await stepAction.importExisting(
           runtime,
           ctx,
           config,
           { currentLabel: stepName, name: def.getName(ctx), version: def.getVersion(ctx) },
           existingKeys
         );
+
```

Comment:
These changes are required for when a partial deployment has an empty step, when running `alter import` this would cause the command to fail.

The idea is it autopopulates the step with an empty value and then importExisting does the rest of the work.